scene/gui/graph_node.cpp

```diff
@@ -139,87 +150,91 @@ void GraphNode::_get_property_list(List<PropertyInfo> *p_list) const {
+	// First pass, determine minimum size AND amount of stretchable elements.
+	Ref<StyleBox> sb_slot = get_theme_stylebox(SNAME("slot"));
+	int separation = get_theme_constant(SNAME("separation"));
 
-	bool first = true;
 	int children_count = 0;
 	int stretch_min = 0;
-	int stretch_avail = 0;
+	int available_stretch_space = 0;
 	float stretch_ratio_total = 0;
 	HashMap<Control *, _MinSizeCache> min_size_cache;
 
-	for (int i = 0; i < get_child_count(); i++) {
-		Control *c = Object::cast_to<Control>(get_child(i));
-		if (!c || !c->is_visible_in_tree()) {
-			continue;
-		}
-		if (c->is_set_as_top_level()) {
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *child = Object::cast_to<Control>(get_child(i, false));
+
+		if (!child || !child->is_visible_in_tree() || child->is_set_as_top_level()) {
 			continue;
 		}
 
-		Size2i size = c->get_combined_minimum_size() + (slot_info[i].draw_stylebox ? sb_slot->get_minimum_size() : Size2());
-		_MinSizeCache msc;
+		Size2i size = child->get_combined_minimum_size() + (slot_table[i].draw_stylebox ? sb_slot->get_minimum_size() : Size2());
 
 		stretch_min += size.height;
+
+		_MinSizeCache msc;
 		msc.min_size = size.height;
-		msc.will_stretch = c->get_v_size_flags().has_flag(SIZE_EXPAND);
+		msc.will_stretch = child->get_v_size_flags().has_flag(SIZE_EXPAND);
+		msc.final_size = msc.min_size;
+		min_size_cache[child] = msc;
 
 		if (msc.will_stretch) {
-			stretch_avail += msc.min_size;
-			stretch_ratio_total += c->get_stretch_ratio();
+			available_stretch_space += msc.min_size;
+			stretch_ratio_total += child->get_stretch_ratio();
 		}
-		msc.final_size = msc.min_size;
-		min_size_cache[c] = msc;
+
 		children_count++;
 	}
 
 	if (children_count == 0) {
 		return;
 	}
 
-	int stretch_max = new_size.height - (children_count - 1) * sep;
+	int stretch_max = new_size.height - (children_count - 1) * separation;
 	int stretch_diff = stretch_max - stretch_min;
-	if (stretch_diff < 0) {
-		//avoid negative stretch space
-		stretch_diff = 0;
-	}
 
-	stretch_avail += stretch_diff - sb->get_margin(SIDE_BOTTOM) - sb->get_margin(SIDE_TOP); //available stretch space.
-	/** Second, pass successively to discard elements that can't be stretched, this will run while stretchable
-		elements exist */
+	// Avoid negative stretch space.
+	stretch_diff = MAX(stretch_diff, 0);
 
-	while (stretch_ratio_total > 0) { // first of all, don't even be here if no stretchable objects exist
-		bool refit_successful = true; //assume refit-test will go well
+	available_stretch_space += stretch_diff - sb_panel->get_margin(SIDE_BOTTOM) - sb_panel->get_margin(SIDE_TOP);
 
-		for (int i = 0; i < get_child_count(); i++) {
-			Control *c = Object::cast_to<Control>(get_child(i));
-			if (!c || !c->is_visible_in_tree()) {
-				continue;
-			}
-			if (c->is_set_as_top_level()) {
+	// Second, pass successively to discard elements that can't be stretched, this will run
+	// while stretchable elements exist.
```

Comment:
```suggestion
	// Second pass, discard elements that can't be stretched, this will run while stretchable elements exist.
```

Reply:
You changed it to be a single line, but that's not what this suggestion was about. 🙂  I proposed different phrasing, which also fixes some weird grammar/punctuation.

Reply:
Whoops, fixed.